File: python/test_base58.py
import unittest
from base58 import *


class TestBase58(unittest.TestCase):
    def test_encode_fixed(self):
        self.assertEqual("11111111111", encode(0))
        self.assertEqual("11111111112", encode(1))
        self.assertEqual("11111111121", encode(58))
        self.assertEqual("2121BWoLqHW", encode(0x05FAFAFAFAFAFAF5))
        self.assertEqual("jpXCZedGfVQ", encode(0xFFFFFFFFFFFFFFFF))

        with self.assertRaises(ValueError):
            encode(-1)

        with self.assertRaises(ValueError):
            encode(0x10000000000000000)

    def test_decode_fixed(self):
        self.assertEqual(0, decode("11111111111"))
        self.assertEqual(1, decode("11111111112"))
        self.assertEqual(58, decode("11111111121"))
        self.assertEqual(0x05FAFAFAFAFAFAF5, decode("2121BWoLqHW"))
        self.assertEqual(0xFFFFFFFFFFFFFFFF, decode("jpXCZedGfVQ"))

        with self.assertRaises(ValueError):
            decode("123456789a")

        with self.assertRaises(ValueError):
            decode("jpXCZedGfVR")

        with self.assertRaises(ValueError):
            decode("zzzzzzzzzzz")

        with self.assertRaises(ValueError):
            decode("11111111!!!")

    def test_encode_decode(self):
        for i in range(0, 0xFFFFFFFFFFFFFFFF, 0x1010101010101):
            self.assertEqual(i, decode(encode(i)))

    def test_max_int_token(self):
        self.assertEqual(encode(max_int), max_token)
        self.assertEqual(decode(max_token), max_int)

File: php/test/TestBase58.php
<?php

require_once __DIR__ . '/../src/Base58.php';

class TestBase58 extends PHPUnit\Framework\TestCase
{
    public function testEncodeFixed()
    {
        $this->assertEquals("11111111111", encode(0));
        $this->assertEquals("11111111112", encode(1));
        $this->assertEquals("11111111121", encode(58));
        $this->assertEquals("2121BWoLqHW", encode(0x05FAFAFAFAFAFAF5));
        $this->assertEquals("NQm6nKp8qFC", encode(9223372036854775807));
        $this->assertEquals("NQm6nKp8qFD", encode(-9223372036854775808));
        $this->assertEquals("jpXCZedGfVQ", encode(-1));
    }

    public function testDecodeFixed()
    {
        $this->assertEquals(0, decode("11111111111"));
        $this->assertEquals(1, decode("11111111112"));
        $this->assertEquals(58, decode("11111111121"));
        $this->assertEquals(0x05FAFAFAFAFAFAF5, decode("2121BWoLqHW"));
        $this->assertEquals(9223372036854775807, decode("NQm6nKp8qFC"));
        $this->assertEquals(-9223372036854775808, decode("NQm6nKp8qFD"));
        $this->assertEquals(-1, decode("jpXCZedGfVQ"));
    }

    public function testEncodeDecode()
    {
        for ($i = PHP_INT_MIN; $i < PHP_INT_MAX; $i += 0x1010101010101) {
            $this->assertEquals($i, decode(encode($i)));
        }
    }
}
